src/raftstore/store/snap.rs

```diff
@@ -61,223 +108,519 @@ impl Display for SnapKey {
     }
 }
 
-/// Name prefix for the self-generated snapshot file.
-const SNAP_GEN_PREFIX: &'static str = "gen";
-/// Name prefix for the received snapshot file.
-const SNAP_REV_PREFIX: &'static str = "rev";
+/// `SnapshotGenerator` is a trait that generates snapshot from `RocksDB` snapshot,
+/// the generated snapshot is used to replicate local db data to other remote raftstores.
+pub trait SnapshotGenerator {
+    fn generate(&mut self,
```

Comment:
Its naming have been changed from builder.build(), creater.create() and then generator.generate(). But it does'n make that difference IMO.   
The names in code will not map into the corresponding design pattern all the time strictly, for example, a few `Manager` in the code base.